lib/workload/stateful/apigw/component.ts

```diff
@@ -0,0 +1,42 @@
+import { Construct } from 'constructs';
+import { aws_ssm, Duration } from 'aws-cdk-lib';
+import * as cognito from 'aws-cdk-lib/aws-cognito';
+import { HttpUserPoolAuthorizer } from 'aws-cdk-lib/aws-apigatewayv2-authorizers';
+import { SHARED_HTTP_API_ID } from '../../../../config/param';
+import { CorsHttpMethod, HttpApi } from 'aws-cdk-lib/aws-apigatewayv2';
+
+export class SharedApiGatewayConstruct extends Construct {
```

Comment:
Is the idea to reuse this for all microservices? For some reason I thought the API gateway would be stateless?

Reply:
Yes, agreed. Maybe refactor it out as a "common" construct that can be reused in (stateless) stacks?

Reply:
Possibly to put the construct directly on the top level of the stateless stack instead?

https://github.com/umccr/orcabus/blob/feature/pipeline-chores/lib/workload/orcabus-stateless-stack.ts#L53-L60

Reply:
Sound good, folks. I will move it back into its own service for now. 

The common construct sound like the plan. But let do that a little bit down the track - until we get to the point where we can collect few use cases to generalise. Then, we shall refactor their commonality from the specificness.

Reply:
Have moved the APIGW into the service boundary. Pls see, if any.